### PR TITLE
scons: update to 4.7.0, drop all the old versions

### PR DESCRIPTION
**Pull request by K900**
## Description of changes

This techinically removes the man pages, but that's probably fine?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by AndersonTorres** [comment]
Are the older versions not needed anymore? Yay!

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -19317,10 +19317,7 @@ with pkgs;
 
   sca2d = callPackage ../development/tools/sca2d {  };
 
-  scons = scons_4_5_2;
-  scons_3_1_2 = callPackage ../development/tools/build-managers/scons/3.1.2.nix { };
-  scons_4_1_0 = callPackage ../development/tools/build-managers/scons/4.1.0.nix { };
-  scons_4_5_2 = callPackage ../development/tools/build-managers/scons/4.5.2.nix { };
+  scons = callPackage ../development/tools/build-managers/scons { };
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:19320]: Migrate this to by-name

File: pkgs/development/tools/build-managers/scons/default.nix [hunk 2]
```diff
@@ -1,28 +1,27 @@
 { lib, fetchFromGitHub, python3 }:
-
-let
+python3.pkgs.buildPythonApplication rec {
```
Comment (AndersonTorres) [inline pkgs/development/tools/build-managers/scons/default.nix:2]: 1. are we finalAttrs yet?
2. python3.pkgs should be changed to python3Packages - or someone will come from the depths of some gehenna to complain about splicing, cross compiling and packages breaking on MIPS32.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/build-managers/scons/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/tools/build-managers/scons/default.nix:2]: 1. No
2. Yes

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (K900) [reply pkgs/top-level/all-packages.nix:19320]: Done

**Comment by mweinelt** [comment]
Same for openvino

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
Next will be how to fix the dreadful lack of manpages

**Comment by doronbehar** [comment]
Hmm I wonder how come CI was green. A quick `git grep scons.override` reveals a few more evaluations fixes missing...

**Comment by K900** [comment]
Those were probably merged into master later.

**Comment by K900** [comment]
I'll do another pass of these after the next staging cycle, probably.

**Comment by doronbehar** [comment]
I already opened https://github.com/NixOS/nixpkgs/pull/326478 to help other PRs that target currently `staging`.

**Pull request merged**